src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/repository/ReportRepository.kt

```diff
@@ -13,6 +17,11 @@ interface ReportRepository : JpaRepository<Report, UUID>, JpaSpecificationExecut
   @EntityGraph(value = "Report.eager", type = EntityGraph.EntityGraphType.LOAD)
   fun findOneEagerlyById(id: UUID): Report?
 
+  @Lock(LockModeType.PESSIMISTIC_WRITE)
+  @QueryHints(value = [QueryHint(name = "jakarta.persistence.lock.timeout", value = "2000")])
+  @Query(value = "SELECT r.id FROM Report r WHERE r.id = :id")
+  fun findReportById(id: UUID): UUID?
```

Comment:
could this method name include something about locking? cos otherwise `reportRepository.findReportById(reportId)` looks so innocuous!

Reply:
yes!

Reply:
Done!

Reply:
and i'm curious to know what sql comes out…

Reply:
`Hibernate: select r1_0.id from report r1_0 where r1_0.id=? for no key update`